android/src/main/java/com/mapbox/mapboxgl/MapboxMapController.java

```diff
@@ -405,8 +408,23 @@ public void onMethodCall(MethodCall call, MethodChannel.Result result) {
         Convert.interpretSymbolOptions(call.argument("options"), symbolBuilder);
         final Symbol symbol = symbolBuilder.build();
         final String symbolId = String.valueOf(symbol.getId());
-        symbols.put(symbolId, new SymbolController(symbol, true, this));
-        result.success(symbolId);
+        Bitmap bitmap = null;
+
+        try {
+          DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
+          bitmap = getImage(symbol, displayMetrics.density);
+        } catch (UnsupportedOperationException e) {
+          Log.e(TAG, "Possible invalid path.");
+          e.printStackTrace();
+        }
+
+        if (bitmap != null) {
+          mapboxMap.getStyle().addImage(symbol.getIconImage(), bitmap);
```

Comment:
style can return null if the underlying style has been fully loaded. This can be worked around by using the async callback version of the method that takes in OnStyleLoaded callback

Reply:
Done. Also simplified the code a bit.

Reply:
Still think we can improve this by adding images when the [OnStyleImageMissingListener](https://docs.mapbox.com/android/api/map-sdk/8.4.0/com/mapbox/mapboxsdk/maps/MapView.OnStyleImageMissingListener.html#onStyleImageMissing-java.lang.String-) fires. This way images can be reused (which is not the case at the moment).

Reply:
Implemented the `addOnStyleImageMissingListener` callback for loading the image.

@tobrun This PR is ready to be merge. Could you review the last changes and merge the PR if you accept the changes?

Reply:
Changes look good but can't merge due to: `This branch has conflicts that must be resolved`. Could you fix that, happy to merge after!

Reply:
@tobrun I've merged master to resolved the conflicts.